### PR TITLE
Allow parquet encryption/decryption keys to be passed in as base64 encoded strings

### PR DESCRIPTION
**Pull request by elefeint**
C++ std::string can contain arbitrary byte arrays, so the change to support arbitrary byte array keys is minimal -- just a way to accept them in Base64.

### CODE REVIEW

File: extension/parquet/parquet_crypto.cpp
Comment (carlopi) [inline extension/parquet/parquet_crypto.cpp:378]: Try catch around Base64Decode to always return:
```
throw InvalidInputException(
			    "Invalid AES key. Not a plain AES key NOR a base64 encoded string")
```
?

I think it's ConversionError is rather weird otherwise.

---

I see this being useful, unsure if it might not worth to have directly the CREATE SECRET parquet (either as part of parquet or directly in 1.0.0 via secondary parquet extension).

File: extension/parquet/parquet_crypto.cpp
Comment (Mytherin) [inline extension/parquet/parquet_crypto.cpp:361]: Can we change this to use `string` instead of `std::string`?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Agreed with @carlopi that the error message isn't very helpful now - plus one more nitpick. Otherwise LGTM

**Review by lnkuiper — APPROVED** [review]
Carlo and Mark have said enough! LGTM from my side once the feedback is implemented.

**Comment by carlopi** [comment]
Thanks! LGTM

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**
